src/core/instance/events.js

```diff
@@ -21,6 +22,8 @@ export function initEvents (vm: Component) {
 
 let target: Component
 
+const updateEventRE = /^update:([\w.$]+)\.([\w$]+)$/
```

Comment:
@javoski  `update:.foo.bar` would still pass through the regex and produce the same matching length

isn't it undesirable, since you're getting the first split chunk? removing the first dot from your expression would deal with this